diff --git a/node_modules/aws-lambda-ws-server/src/handler.js b/node_modules/aws-lambda-ws-server/src/handler.js
index 77eadab..0cfd7a2 100644
--- a/node_modules/aws-lambda-ws-server/src/handler.js
+++ b/node_modules/aws-lambda-ws-server/src/handler.js
@@ -14,12 +14,7 @@ module.exports = routes => async (event, context) => {
   } else if (eventType === 'DISCONNECT' && routes.disconnect) {
     return routes.disconnect(connectionArgs)
   } else if (eventType === 'MESSAGE') {
-    const body = JSON.parse(
-      Buffer.from(
-        event.body || '{}',
-        event.isBase64Encoded ? 'base64' : undefined
-      )
-    )
+    const body = event.body
     const messageArgs = { ...connectionArgs, message: body }
     if (routes[routeKey]) {
       return routes[routeKey](messageArgs)
diff --git a/node_modules/aws-lambda-ws-server/src/local.js b/node_modules/aws-lambda-ws-server/src/local.js
index 3d3cf12..f4c58ec 100644
--- a/node_modules/aws-lambda-ws-server/src/local.js
+++ b/node_modules/aws-lambda-ws-server/src/local.js
@@ -89,7 +89,7 @@ const context = () => ({
         err.statusCode = 410
         return reject(err)
       }
-      ws.send(JSON.stringify(payload), err => {
+      ws.send(payload, err => {
         if (err) return reject(err)
         resolve()
       })
@@ -123,7 +123,7 @@ module.exports = handler => {
     })
     ws.on('message', async message => {
       try {
-        const body = JSON.parse(message || '{}')
+        const body = message
         await handler(
           event(body[mappingKey] || '$default', 'MESSAGE', req, message),
           context()
